GitTfsTest/Integration/CloneTests.cs

```diff
@@ -387,5 +388,45 @@ public void CloneWithAllBranchesShouldHandleFolderDeletedAndRecreatedAsBranch()
             };
             AssertNewClone("MyTeamProject", refs);
         }
+
+
+        [FactExceptOnUnix]
+        public void CloneAllBranchesShouldHandleMergeFromRenameChangeset()
+        {
+            h.SetupFake(r =>
+            {
+                r.SetRootBranch("$/MyTeamProject/Root");
+
+                r.Changeset(1, "Create root branch", DateTime.Now)
+                    .Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyTeamProject/Root")
+                    .Change(TfsChangeType.Add, TfsItemType.File, "$/MyTeamProject/Root/File.txt", new byte[0], itemId: 100);
+
+                r.BranchChangeset(2, "Create another branch", DateTime.Now, "$/MyTeamProject/Root", "$/MyTeamProject/Branch", 1)
+                    .Change(TfsChangeType.Branch, TfsItemType.Folder, "$/MyTeamProject/Branch")
+                    .Change(TfsChangeType.Branch, TfsItemType.File, "$/MyTeamProject/Branch/File.txt", new byte[0], itemId: 100);
+
+                r.BranchChangeset(3, "Rename other branch", DateTime.Now, "$/MyTeamProject/Branch", "$/MyTeamProject/BranchRenamed", 2)
+                    .Change(TfsChangeType.Rename, TfsItemType.Folder, "$/MyTeamProject/BranchRenamed")
+                    .Change(TfsChangeType.Delete, TfsItemType.Folder, "$/MyTeamProject/Branch")
+                    .Change(TfsChangeType.Rename, TfsItemType.File, "$/MyTeamProject/BranchRenamed/File.txt", new byte[0], itemId: 100)
+                    .Change(TfsChangeType.Delete, TfsItemType.File, "$/MyTeamProject/Branch/File.txt");
```

Comment:
Great, that makes more sense! I I had been copying this file via Find All References on `TfsChangeType.Rename`:
https://github.com/git-tfs/git-tfs/blob/master/GitTfsTest/Fixtures/vtccds.cs#L62-L64
https://github.com/git-tfs/git-tfs/blob/master/GitTfsTest/Fixtures/vtccds.cs#L49-L53
etc.